src/storage-types/src/read_holds.rs

```diff
@@ -83,18 +96,14 @@ impl<T: TimelyTimestamp> ReadHold<T> {
     /// Merges `other` into `self`, keeping the overall read hold.
     ///
     /// # Panics
+    ///
     /// Panics when trying to merge a [ReadHold] for a different collection
-    /// (different [GlobalId]) or when trying to merge a [ReadHold] from a
-    /// different issuer.
+    /// (different [GlobalId]).
     pub fn merge_assign(&mut self, mut other: ReadHold<T>) {
         assert_eq!(
             self.id, other.id,
             "can only merge ReadHolds for the same ID"
         );
-        assert!(
-            self.holds_tx.same_channel(&other.holds_tx),
-            "can only merge ReadHolds that come from the same issuer"
-        );
```

Comment:
If anybody feels strongly about keeping this check, we could make `ChangeTx` a custom trait, e.g.:

```
trait ChangeTx {
    fn send(&self, id: GlobalId, change: ChangeBatch) -> Result;
    fn same_recipient(&self, other: &Self) -> bool;
}
```

Reply:
You can do this without resorting to a trait. Effectively what you want to do is create a single prototypical `Arc<sender closure>` in the controller that gets cloned into all read holds that are given out.  Then this assert can be replaced with `Arc::ptr_eq`

Reply:
Ah, true. That would still allow people to create `ReadHold`s that write to the same channel but use different closures for it, but we can hope that CI would catch this.

Reply:
Done

Reply:
Oh, actually I think we should probably not do this. With this scheme, two `ReadHold`s created with `with_channel` for the same channel won't be mergeable. So if you request two read holds for the same collection from the storage controller, then try to merge them, Mz will panic. CI suggests that we don't try to do such things today, but we might in edge-cases that are not covered by CI, in which case the assert would be a time bomb waiting to explode in production.